protocol/v2/ssv/runner/runner.go

```diff
@@ -60,14 +63,22 @@ type BaseRunner struct {
 	TimeoutF TimeoutF `json:"-"`
 }
 
-func NewBaseRunner(logger *zap.Logger) *BaseRunner {
+func NewBaseRunner(logger *zap.Logger, beaconRoleType spectypes.BeaconRole, beaconNetwork spectypes.BeaconNetwork, share *spectypes.Share, qbftController *controller.Controller) *BaseRunner {
 	return &BaseRunner{
-		logger: logger,
+		mtx:            &sync.RWMutex{},
+		logger:         logger.With(zap.String("who", "BaseRunner")),
+		BeaconRoleType: beaconRoleType,
+		BeaconNetwork:  beaconNetwork,
+		Share:          share,
+		QBFTController: qbftController,
 	}
 }
 
 // baseStartNewDuty is a base func that all runner implementation can call to start a duty
 func (b *BaseRunner) baseStartNewDuty(runner Runner, duty *spectypes.Duty) error {
+	b.mtx.Lock() // writes to b.State
+	defer b.mtx.Unlock()
+
```

Comment:
Try to lock, ask canStartNewDuty & assign the new state, and then unlock (before executeDuty which is a heavy call [uses io])